src/Microsoft.Diagnostics.EventFlow.Core/Implementations/DiagnosticPipeline.cs

```diff
@@ -229,16 +290,19 @@ public EventData[] FilterAndDecorate(EventData[] eventsToFilter)
                     return null;
                 }
 
-                // CONSIDER: using a Queue<T> with external lock and be able to reuse the queue
-                // (ConcurrentQueue has no Clear() method while ordinary Queue does).
-                ConcurrentQueue<EventData> eventsToKeep = new ConcurrentQueue<EventData>();
+                ConcurrentStack<EventData> eventsToKeep;
```

Comment:
Why switch from queue to stack? With the switch, the events are now returned in reverse order in the array. Does it matter?